Thumbs/ThumbsMiddleware.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Threading.Tasks;
+using DLCS.Model.Assets;
+using DLCS.Model.Storage;
+using DLCS.Web.Requests.AssetDelivery;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
+using Newtonsoft.Json;
+
+namespace Thumbs
+{
+    public class ThumbsMiddleware
+    {
+        private readonly int cacheSeconds;
+        private readonly IConfiguration configuration;
+        private readonly ILogger<ThumbsMiddleware> logger;
+        private readonly IThumbRepository thumbRepository;
+
+        public ThumbsMiddleware(
+            RequestDelegate next,
+            IConfiguration configuration,
+            ILogger<ThumbsMiddleware> logger,
+            IThumbRepository thumbRepository)
+        {
+            this.configuration = configuration;
+            this.cacheSeconds = configuration.GetValue<int>("responseCacheSeconds", 0);
+            this.logger = logger;
+            this.thumbRepository = thumbRepository;
+        }
+
+        public async Task Invoke(HttpContext context,
+            IBucketReader bucketReader,
+            AssetDeliveryPathParser parser)
+        {
+            var thumbnailRequest = parser.Parse(context.Request.Path.Value);
+            if (thumbnailRequest.IIIFImageRequest.IsBase)
+            {
+                await RedirectToInfoJson(context);
+            }
+            else if (thumbnailRequest.IIIFImageRequest.IsInformationRequest)
+            {
+                await WriteInfoJson(context, thumbnailRequest);
+            }
+            else
+            {
+                // mode for debugging etc
+                switch (context.Request.Query["mode"])
+                {
+                    case "dump":
+                        await WriteRequestDump(context, thumbnailRequest);
+                        break;
+                    default:
```

Comment:
Should we disable this in production? Or is it going to be useful? Anything in the context that would be suitable for responding with?

Reply:
probably useful in prod. Shows how the image request was parsed.